app/scripts/services/tile-proxy.js

```diff
@@ -34,76 +34,50 @@ export let authHeader = null;
  */
 
 /**
+ * Create a function that batches calls at intervals, with a final debounce.
+ *
+ * The returned function collects individual items and executes `processBatch` at the specified interval.
+ * If additional calls occur after the last batch, a final debounce ensures they are included.
+ *
+ * @template T
  * @template {Array<unknown>} Args
  *
- * @param {(ctx: RequestContext, ...args: Args) => void} func
+ * @param {(items: Array<T>, ...args: Args) => void} processBatch
  * @param {number} interval
  * @param {number} finalWait
  */
-const throttleAndDebounce = (func, interval, finalWait) => {
+const delayedBatchExecutor = (processBatch, interval, finalWait) => {
   /** @type {ReturnType<typeof setTimeout> | undefined} */
   let timeout = undefined;
-  /** @type {Array<WrappedTilesRequest>} */
-  let bundledRequest = [];
-  /** @type {Record<string, number>} */
-  let requestMapper = {};
+  /** @type {Array<T>} */
+  let items = [];
   /** @type {number} */
   let blockedCalls = 0;
 
-  /**
-   * @param {WrappedTilesRequest} request
-   * @returns {void}
-   */
-  const bundleRequests = (request) => {
-    const requestId = requestMapper[request.id];
-
-    if (requestId && bundledRequest[requestId]) {
-      bundledRequest[requestId].ids = bundledRequest[requestId].ids.concat(
-        request.ids,
-      );
-    } else {
-      requestMapper[request.id] = bundledRequest.length;
-      bundledRequest.push(request);
-    }
-  };
-
   const reset = () => {
     timeout = undefined;
-    bundledRequest = [];
-    requestMapper = {};
+    items = [];
   };
 
-  /**
-   * @param {WrappedTilesRequest} _request
-   * @param {Args} args
-   */
-  const callFunc = (_request, ...args) => {
+  /** @param {Args} args */
+  const callFunc = (...args) => {
     // NB: In a normal situation we would just call `func(...args)` but since we
-    // modify the first argument and always trigger `reset()` afterwards I created
+    // always trigger `reset()` afterwards I created
     // this helper function to avoid code duplication. Think of this function
     // as the actual function call that is being throttled and debounced.
-    func(
-      {
-        sessionId,
-        requests: bundledRequest,
-      },
-      ...args,
-    );
+    processBatch(items, ...args);
```

Comment:
we now just submit the "bundle" (items) to a processor when either debouced or throttled.